toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -394,3 +402,70 @@ func addCustomizerRelease(imageChroot *safechroot.Chroot, toolVersion string, bu
 
 	return nil
 }
+
+func handleSELinux(selinuxMode imagecustomizerapi.SELinux, selinuxDisablePackageCheck bool,
+	partitionsCustomized bool, imageChroot *safechroot.Chroot,
+) error {
+	var err error
+
+	// Resolve the default SELinux mode.
+	if selinuxMode == imagecustomizerapi.SELinuxDefault {
+		selinuxMode, err = getCurrentSELinuxMode(imageChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If the partitions were customized, then the grub.cfg file will have been recreated from scratch and therefore
+	// the SELinux args will already be correct and don't need to be updated.
+	if !partitionsCustomized {
+		// Update the SELinux kernel command-line args.
+		err := updateSELinuxCommandLine(selinuxMode, imageChroot)
+		if err != nil {
+			return fmt.Errorf("failed to update SELinux args in grub.cfg:\n%w", err)
+		}
+	}
+
+	if selinuxMode != imagecustomizerapi.SELinuxDisabled {
+		imagerSELinuxMode, err := selinuxModeToImager(selinuxMode)
+		if err != nil {
+			return err
+		}
+
+		if !selinuxDisablePackageCheck {
+			// Check if the selinux-policy package is installed.
+			err = imageChroot.UnsafeRun(func() error {
+				stdout, _, err := shell.Execute("rpm", "-q", configuration.SELinuxPolicyDefault)
+				if err != nil {
+					// Note: RPM's stdout will say somelike like "package selinux-policy is not installed".
+					stdout = strings.TrimSpace(stdout)
+					return fmt.Errorf("SELinux is enabled so the policy package must be installed:\n%s:\n%w", stdout, err)
+				}
```

Comment:
Suggestion: it might be worth to extract bits of code into well-named functions, if you feel they are complex enough they require a comment. That way the comment is encapsulated in the function's name and the intention/flow of the program may be easier to follow.